src/main/java/model/structures/GraphImpl.java

```diff
@@ -0,0 +1,144 @@
+package model.structures;
+
+import model.generator.ModelGenerator;
+
+import java.util.*;
+
+public class GraphImpl implements Graph {
+
+    private HashMap<Integer, Node> nodes;
+    private int[][] adjacencyMatrix;
+    private Random randomForAdjacencyMatrix;
+    private int edgeWageSum;
+    private int graphSize;
+    private int iterationStep;
+    private final long defaultSeedForWages = 8064995938733697569L;
+
+    public GraphImpl(long seedForWages, long seedForHabitats, int edgeWageSum, int minimalNumberOfHabitats, int maximumNumberOfHabitats, int graphSize, int infectingParameter)
+    {
+        this.graphSize = graphSize;
+        this.edgeWageSum = edgeWageSum;
+        randomForAdjacencyMatrix = new Random(seedForWages);
+        adjacencyMatrix = ModelGenerator.generateAdjacencyMatrix(graphSize, edgeWageSum, randomForAdjacencyMatrix);
+        generatePeople(seedForHabitats, minimalNumberOfHabitats, maximumNumberOfHabitats, graphSize, infectingParameter);
+        iterationStep = 0;
+    }
+
+    private GraphImpl(int[][] adjacencyMatrix, HashMap<Integer, Node> nodes)
+    {
+        this.adjacencyMatrix = adjacencyMatrix;
+        this.nodes = nodes;
+        randomForAdjacencyMatrix = new Random();
+    }
+
+    public void setIterationStep(int iterationStep)
+    {
+        this.iterationStep = iterationStep;
+    }
+
+    public int getIterationStep()
+    {
+        return iterationStep;
+    }
+
+    public GraphImpl(ArrayList<Node> nodeArrayList, int[][] adjacencyMatrix)
+    {
+        this.adjacencyMatrix = adjacencyMatrix;
+        graphSize = nodeArrayList.size();
+
+        nodes = new HashMap<>();
+        edgeWageSum = 0;
+
+        for(Node node : nodeArrayList)
+            nodes.put(node.getId(), node);
+
+        for (int[] matrix : adjacencyMatrix)
+            for (int j = 0; j < adjacencyMatrix.length; j++)
+                edgeWageSum += matrix[j];
+
+        edgeWageSum /= 2;
+
```

Comment:
W tym konstruktorze jest bug. Nie jest tworzony Random. Żeby uniknąć takich błedów, lepiej żeby konstruktor wołał inny konstruktor zamiast robić to samo trzy razy

Reply:
Rzeczywiście, pragnąłbym zauważyć, że jednak logika i przypadki użycia pozostałych są różne, mądrzejsze byłoby przekazanie random tu jako parametr

Reply:
w sensie nie wiem czym jest random tylko wiem że się wywalało jak odpalałam tym konstruktorem, po prostu zrób żeby było dobrze

Reply:
Hej co z tymi konstruktorami?